### DISCUSSION TITLE
How to use VSCodium with Wayland?

### DISCUSSION THREAD

**Discussion by jaroshevskii** [VSCodium/vscodium#1381] [Q&A] [answered]
If you run `vscodium` from a terminal with `--ozone-platform=auto` flag.  Everything works fine (only the icon is not displayed). 

The problem is how to make vscodium automatically read this flag.  In normal VSCode, I could just specify the flag in `code-flags.conf` (in my case, it's in `~/.config` folder).  For some reason this doesn't work for VSCodium or I'm dumb somewhere.

**Comment by paulcarroty** [comment]
AFAIR VSCode does not support configs in `~/.config`, `/etc` or any location excluding VSC src dir maybe. So we're forced to tune `.desktop` for GUI and use `alias` for cli.

**Comment by noonsleeper** [comment]
Hi @jaroshevskii you can use a solution like [this](https://github.com/flathub/com.vscodium.codium/issues/162), override the original `.desktop` for your user.

**Reply by jaroshevskii** [reply]
Thanks for the advice. I will look into this method

**Comment by paulcarroty** [comment]
Tuning the `.desktop` with `--ozone-platform-hint=auto` works for me.

**Reply by jaroshevskii** [reply]
Thanks for the advice. I will look into this method

**Reply by karlpokus** [reply]
Yup this works. But the font seems changed. Wider?

**Comment by mvirts** [comment]
Just ran into this issue and auto didn't work for me, but `--ozone-platform-hint=wayland` did.

**Reply by itsntme1** [reply]
Thanks, worked. I found that `--ozone-platform=wayland` doesn't display the warning.

**Comment by trinitronx** [comment, marked as answer]
> In normal VSCode, I could just specify the flag in `code-flags.conf`

This config file seems to be specific to ["`Code - OSS`" on ArchLinux][1], and is read by a wrapper script. In that `code` package, `/usr/bin/code` is a shell script which reads `code-flags.conf` and `electronNN-flags.conf`, then `exec`s the real `electron` binary with those flags.

[1]: https://wiki.archlinux.org/title/Visual_Studio_Code
